client/src/views/Settings/Tabs/EditPrivacy/index.vue

```diff
@@ -70,26 +60,191 @@
           <q-item-label>Hide offensive comments</q-item-label>
           <q-item-label caption>Hide comments</q-item-label>
         </q-item-section>
-        <q-item-section side >
+        <q-item-section side>
           <q-toggle color="blue" v-model="hideOffencive" />
         </q-item-section>
       </q-item>
 
+      <q-separator spaced />
+      <q-item-label header>Black list</q-item-label>
+      <div class="row justify-between">
+        <div id="input-search" ref="search" class="col">
+          <q-input v-model="nickname" @input="searchByNick" />
+          <block-search
+            v-if="showSearch"
+            id="search-result"
+            :results="nicknameSearchResults"
+            @handleChoose="handleChoose"
+          />
+        </div>
+        <q-btn @click="blockUser" id="block-button" class="col" :disabled="!enableBlock">Block</q-btn>
+      </div>
+      <p v-if="userToBlockNick">
+        {{userToBlockNick}}
+        <q-btn round color="teal" size="xs" icon="close" @click="userToBlockNick = ''" />
+      </p>
+      <p id="error-message" v-if="errorMessage">
+        {{errorMessage}}
+        <q-btn round color="red" size="xs" icon="close" @click="errorMessage = ''" />
+      </p>
+      <blocked-list id="blocked-list" @unblock="unblockUser" :list="blockedProfilesList"></blocked-list>
     </q-list>
   </div>
 </template>
 
 <script>
+import BlockSearch from '@/components/BlockSearch.vue'
+import BlockedList from '@/components/BlockedList.vue'
+import { debounce } from 'lodash'
+import { searchByNick, profileAction, getProfile } from '@/services/profile.js'
+import { mapGetters } from 'vuex'
+
 export default {
   name: 'EditPrivacy',
+  components: {
+    BlockSearch,
+    BlockedList
+  },
+  computed: {
+    ...mapGetters({
+      myid: 'profile/myProfileId'
+    })
+  },
   data () {
     return {
       comment: true,
       like: true,
       reply: true,
       uploadPost: true,
-      hideOffencive: false
+      hideOffencive: false,
+      nickname: '',
+      showSearch: true,
+      nicknameSearchResults: [],
+      userToBlockNick: '',
+      userToBlockId: '',
+      errorMessage: '',
+      enableBlock: false,
+      blockedIdList: [],
+      blockedProfilesList: []
+    }
+  },
+  created () {
+    this.displayBlockedIds()
+  },
+  methods: {
+    searchByNick: debounce(function (value) {
+      if (value !== '') {
+        this.nicknameSearchResults = []
+        this.showSearch = true
+        searchByNick({ nickname: value })
+          .then(res => {
+            this.nicknameSearchResults = []
+            res.data.message.forEach(element => {
+              this.nicknameSearchResults.push({ id: element.userId, nickname: element.nickname, avatar: element.avatarUrl })
+            })
+          })
+          .catch(err => {
+            if (err.response.data.error === 'No user has been found') {
```

Comment:
Do not rely on messaging(I've already told about it) because the message could be changed.